### PR TITLE
Feature/initial commit

### PR DESCRIPTION
**Pull request by BelehovEgor**

### CODE REVIEW

File: src/OfficeService.Data.Provider/IDataProvider.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using LT.DigitalOffice.Kernel.Database;
+using LT.DigitalOffice.Kernel.Enums;
+using Microsoft.EntityFrameworkCore;
+
+namespace LT.DigitalOffice.OfficeService.Data.Provider
+{
+    [AutoInject(InjectType.Scoped)]
+    public interface IDataProvider : IBaseDataProvider
+    {
+        DbSet<DbOffice> Offices { get; set; }
```
Comment (Cheephot) [inline src/OfficeService.Data.Provider/IDataProvider.cs]: tab

File: src/OfficeService.Mappers/Models/Interfaces/IOfficeInfoMapper.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using LT.DigitalOffice.OfficeService.Models.Db;
+using LT.DigitalOffice.OfficeService.Models.Dto.Models;
+using LT.DigitalOffice.Kernel.Attributes;
+
+namespace LT.DigitalOffice.OfficeService.Mappers.Models.Interfaces
+{
+    [AutoInject]
+    public interface IOfficeInfoMapper
+    {
+        OfficeInfo Map(DbOffice office);
```
Comment (Cheephot) [inline src/OfficeService.Mappers/Models/Interfaces/IOfficeInfoMapper.cs]: tab

File: src/OfficeService.Models.Db/DbOfficeUser.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+
+namespace LT.DigitalOffice.OfficeService.Models.Db
+{
+    public class DbOfficeUser
+    {
+        public const string TableName = "OfficesUsers";
```
Comment (Cheephot) [inline src/OfficeService.Models.Db/DbOfficeUser.cs]: fix tabs

File: src/OfficeService.Models.Dto/OfficeService.Models.Dto.csproj [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+  <ItemGroup>
```
Comment (Cheephot) [inline src/OfficeService.Models.Dto/OfficeService.Models.Dto.csproj]: remove this

File: src/OfficeService.Models.Dto/Requests/Users/ChangeOfficeRequest.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace LT.DigitalOffice.OfficeService.Models.Dto.Requests.Users
+{
+  public record ChangeOfficeRequest
```
Comment (Cheephot) [inline src/OfficeService.Models.Dto/Requests/Users/ChangeOfficeRequest.cs]: ChangeUserOfficeRequest?

File: src/OfficeService.Validation/Office/EditOfficeRequestValidator.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        {
+          { x => !string.IsNullOrEmpty(x.value?.ToString()), "City cannot be empty." },
+        });
+
+      AddFailureForPropertyIf(
+        nameof(EditOfficeRequest.Address),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => !string.IsNullOrEmpty(x.value?.ToString()), "Address cannot be empty." },
```
Comment (Cheephot) [inline src/OfficeService.Validation/Office/EditOfficeRequestValidator.cs]: Trim

**Review by Cheephot — APPROVED** [review]

File: src/OfficeService.Validation/Office/EditOfficeRequestValidator.cs (hunk 6 above)
Comment (BelehovEgor) [reply src/OfficeService.Validation/Office/EditOfficeRequestValidator.cs]: fixed

File: src/OfficeService.Models.Dto/Requests/Users/ChangeOfficeRequest.cs (hunk 5 above)
Comment (BelehovEgor) [reply src/OfficeService.Models.Dto/Requests/Users/ChangeOfficeRequest.cs]: fixed

File: src/OfficeService.Models.Dto/OfficeService.Models.Dto.csproj (hunk 4 above)
Comment (BelehovEgor) [reply src/OfficeService.Models.Dto/OfficeService.Models.Dto.csproj]: removed

File: src/OfficeService.Models.Db/DbOfficeUser.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/OfficeService.Models.Db/DbOfficeUser.cs]: fixed

File: src/OfficeService.Mappers/Models/Interfaces/IOfficeInfoMapper.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/OfficeService.Mappers/Models/Interfaces/IOfficeInfoMapper.cs]: fixed

File: src/OfficeService.Data.Provider/IDataProvider.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/OfficeService.Data.Provider/IDataProvider.cs]: fixed

File: src/OfficeService.Business/Commands/Users/ChangeOfficeCommand.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      _responseCreater = responseCreater;
+      _validator = validator;
+      _mapper = mapper;
+      _repository = repository;
+      _cacheNotebook = cacheNotebook;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(ChangeUserOfficeRequest request)
+    {
+      if (_httpContextAccessor.HttpContext.GetUserId() != request.UserId
```
Comment (MariyaSin) [inline src/OfficeService.Business/Commands/Users/ChangeOfficeCommand.cs:51]: i am not sure what user can change office

File: src/OfficeService.Data.Provider.MsSql.Ef/OfficeServiceDbContext.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  /// </summary>
+  public class OfficeServiceDbContext : DbContext, IDataProvider
+  {
+    public OfficeServiceDbContext(DbContextOptions<OfficeServiceDbContext> options)
+      : base(options)
+    {
+    }
+
+    public DbSet<DbOffice> Offices { get; set; }
+    public DbSet<DbOfficeUser> OfficeUsers { get; set; }
```
Comment (MariyaSin) [inline src/OfficeService.Data.Provider.MsSql.Ef/OfficeServiceDbContext.cs]: OfficesUsers

File: src/OfficeService.Data.Provider/IDataProvider.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+using LT.DigitalOffice.OfficeService.Models.Db;
+using Microsoft.EntityFrameworkCore;
+
+namespace LT.DigitalOffice.OfficeService.Data.Provider
+{
+  [AutoInject(InjectType.Scoped)]
+  public interface IDataProvider : IBaseDataProvider
+  {
+    DbSet<DbOffice> Offices { get; set; }
+    DbSet<DbOfficeUser> OfficeUsers { get; set; }
```
Comment (MariyaSin) [inline src/OfficeService.Data.Provider/IDataProvider.cs]: OfficesUsers

File: src/OfficeService.Data/Interfaces/IOfficeRepository.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  [AutoInject]
+  public interface IOfficeRepository
+  {
+    Task CreateAsync(DbOffice office);
+
+    Task<DbOffice> GetAsync(Guid officeId);
+
+    Task<(List<DbOffice>, int totalCount)> FindAsync(OfficeFindFilter filter);
+
+    Task<bool> EditAsync(Guid officeId, JsonPatchDocument<DbOffice> news);
```
Comment (MariyaSin) [inline src/OfficeService.Data/Interfaces/IOfficeRepository.cs]: news????

File: src/OfficeService.Data/Interfaces/IOfficeUserRepository.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+using LT.DigitalOffice.Kernel.Attributes;
+
+namespace LT.DigitalOffice.OfficeService.Data.Interfaces
+{
+  [AutoInject]
+  public interface IOfficeUserRepository
+  {
+    Task<bool> CreateAsync(DbOfficeUser user);
+
+    Task<List<DbOfficeUser>> GetAsync(List<Guid> userIds);
```
Comment (MariyaSin) [inline src/OfficeService.Data/Interfaces/IOfficeUserRepository.cs]: usersIds

File: src/OfficeService.Models.Db/DbOffice.cs [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    {
+      builder
+        .ToTable(DbOffice.TableName);
+
+      builder
+        .HasKey(o => o.Id);
+
+      builder
+        .Property(o => o.City)
+        .IsRequired();
```
Comment (MariyaSin) [inline src/OfficeService.Models.Db/DbOffice.cs:44]: in migration the property is nullable

File: src/OfficeService.Models.Db/DbOffice.cs [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+
+      builder
+        .Property(o => o.Address)
+        .IsRequired();
```
Comment (MariyaSin) [inline src/OfficeService.Models.Db/DbOffice.cs:48]: in migration the property is nullable

File: src/OfficeService.Models.Dto/Requests/Office/EditOfficeRequest.cs [hunk 14]
```diff
@@ -0,0 +1,11 @@
+﻿namespace LT.DigitalOffice.OfficeService.Models.Dto.Requests.Office
+{
+  public record EditOfficeRequest
+  {
+    public string Name { get; set; }
+    public string City { get; set; }
+    public string Address { get; set; }
+    public double Latitude { get; set; }
+    public double Longitude { get; set; }
```
Comment (MariyaSin) [inline src/OfficeService.Models.Dto/Requests/Office/EditOfficeRequest.cs:9]: IsActive?

File: src/OfficeService/appsettings.json [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+
+    // CRService
+    "CheckUserRightsEndpoint": "CRService_CheckUserRights",
+
+    // AuthService
+    "ValidateTokenEndpoint": "AuthService_ValidateToken",
+
+    // OfficeService receiving endpoints
+    "GetOfficesEndpoint": "OfficeService_GetOffices",
+    "CreateUserOfficeEndpoint": "OfficeService_EditUserOffice"
```
Comment (MariyaSin) [inline src/OfficeService/appsettings.json]: Edit??????

**Review by MariyaSin — APPROVED** [review]

File: src/OfficeService.Data.Provider.MsSql.Ef/OfficeServiceDbContext.cs (hunk 8 above)
Comment (BelehovEgor) [reply src/OfficeService.Data.Provider.MsSql.Ef/OfficeServiceDbContext.cs]: fixed

File: src/OfficeService.Data.Provider/IDataProvider.cs (hunk 9 above)
Comment (BelehovEgor) [reply src/OfficeService.Data.Provider/IDataProvider.cs]: fixed

File: src/OfficeService.Data/Interfaces/IOfficeRepository.cs (hunk 10 above)
Comment (BelehovEgor) [reply src/OfficeService.Data/Interfaces/IOfficeRepository.cs]: fixed

File: src/OfficeService.Data/Interfaces/IOfficeUserRepository.cs (hunk 11 above)
Comment (BelehovEgor) [reply src/OfficeService.Data/Interfaces/IOfficeUserRepository.cs]: fixed

File: src/OfficeService.Models.Db/DbOffice.cs (hunk 12 above)
Comment (BelehovEgor) [reply src/OfficeService.Models.Db/DbOffice.cs:44]: fixed

File: src/OfficeService.Models.Db/DbOffice.cs (hunk 13 above)
Comment (BelehovEgor) [reply src/OfficeService.Models.Db/DbOffice.cs:48]: fixed

File: src/OfficeService.Models.Dto/Requests/Office/EditOfficeRequest.cs (hunk 14 above)
Comment (BelehovEgor) [reply src/OfficeService.Models.Dto/Requests/Office/EditOfficeRequest.cs:9]: added and added to request and validator

File: src/OfficeService/appsettings.json (hunk 15 above)
Comment (BelehovEgor) [reply src/OfficeService/appsettings.json]: renamed

File: src/OfficeService.Data.Provider.MsSql.Ef/OfficeServiceDbContext.cs [hunk 16]
```diff
@@ -0,0 +1,54 @@
+﻿using System.Reflection;
+using System.Threading.Tasks;
+using LT.DigitalOffice.OfficeService.Models.Db;
+using Microsoft.EntityFrameworkCore;
+
+namespace LT.DigitalOffice.OfficeService.Data.Provider.MsSql.Ef
+{
+  /// <summary>
+  /// A class that defines the tables and its properties in the database of OfficeService.
```
Comment (VLiamin) [inline src/OfficeService.Data.Provider.MsSql.Ef/OfficeServiceDbContext.cs:9]: Do we need this comment?

**Review by VLiamin — APPROVED** [review]

File: src/OfficeService.Data.Provider.MsSql.Ef/OfficeServiceDbContext.cs (hunk 16 above)
Comment (BelehovEgor) [reply src/OfficeService.Data.Provider.MsSql.Ef/OfficeServiceDbContext.cs:9]: it is documentation, so I decided don't remove this

**Pull request merged**